File: apple/Sources/ReactNativeWatchOS/Modifiers/Interpolation.swift
import SwiftUI

/// SwiftUI `.interpolation(_:)` — Image-only. Controls how the image is
/// resampled when drawn at a non-native size. `.none` gives crisp pixel
/// art; `.high` is the default-ish smooth fit.
enum RNWInterpolationModifier {
    @MainActor
    static func register(into r: RNWImageModifierRegistry) {
        r.register("interpolation") { image, params in
            switch params.string("level") {
            case "none":   return image.interpolation(.none)
            case "low":    return image.interpolation(.low)
            case "medium": return image.interpolation(.medium)
            case "high":   return image.interpolation(.high)
            default:       return image
            }
        }
    }
}

File: src/modifiers/interpolation.ts
import { createModifier } from './createModifier';

export type InterpolationLevel = 'none' | 'low' | 'medium' | 'high';

/// SwiftUI `.interpolation(_:)`. Image-only — controls how the image is
/// resampled when drawn at a non-native size. `'none'` for crisp pixel
/// art; `'high'` for smooth.
export function interpolation(level: InterpolationLevel) {
  return createModifier('interpolation', { level });
}
